pkg/libdevfile/component.go

```diff
@@ -0,0 +1,44 @@
+package libdevfile
+
+import (
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/library/pkg/devfile/parser"
+	"github.com/devfile/library/pkg/devfile/parser/data/v2/common"
+)
+
+type component interface {
+	CheckValidity() error
+	Apply(handler Handler) error
+}
+
+// newComponent creates a concrete component, based on its type
+func newComponent(devfileObj parser.DevfileObj, devfileCmp v1alpha2.Component) (component, error) {
+	var cmp component
+
+	componentType, err := common.GetComponentType(devfileCmp)
+	if err != nil {
+		return nil, err
+	}
+	switch componentType {
+
+	case v1alpha2.ContainerComponentType:
+		cmp = newContainerComponent(devfileObj, devfileCmp)
+
+	case v1alpha2.KubernetesComponentType:
+		cmp = newKubernetesComponent(devfileObj, devfileCmp)
+
+	case v1alpha2.OpenshiftComponentType:
+		cmp = newOpenshiftComponent(devfileObj, devfileCmp)
```

Comment:
IIRC, OpenShift component type is going to be deprecated and only Kubernetes component type will be available in devfile. I can't find an issue for it in devfile API repo. But @kadel could validate me.

Reply:
I try to be exhaustive in the hope it is included in the devfile library. We coul remove it easily when devfile does not support it anymore